src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -214,6 +214,12 @@ private[snowflake] class JDBCWrapper {
   private def isQuoted(name: String): Boolean = {
     name.startsWith("\"") && name.endsWith("\"") 
   }
+  private def quotedName(name: String): String = {
+      if (name.matches("[_A-Z]([_0-9A-Z])*"))
+        "\"" + name.toUpperCase + "\""
```

Comment:
Woops, I think I messed up the logic here. We shouldn't use the same regex because this is going the other direction (for the Snowflake-> Spark case, we perform the check on names for which the upper-case promotion has already been performed, but here we start from any user-given column names). 

When starting from Spark we instead need to check every unquoted identifier (here in `quotedNames`) to see if it contains any characters that wouldn't otherwise be legal without being quoted. If so, then we simply add the quotes. If it's all legal, then we upper-case and quote. So we need another regex check for name legality. I think this basically amounts to changing the pattern to also include lowercase letters.

See here:
https://docs.snowflake.net/manuals/sql-reference/identifiers.html

> Identifiers that are not enclosed by double quotes are case-insensitive and must start with a letter or the underscore (_) character. In addition, they can only contain a combination of alphanumeric characters and the underscore character.

I hope this makes sense.